src/lib/dependencies/index.js

```diff
@@ -1,73 +1,128 @@
-import { flatten } from 'lodash';
+import cache from '../cache';
+import debug from 'debug';
+import { fetchFileFromRepo, searchFilesFromRepo } from '../github';
+import minimatch from 'minimatch';
 
 import * as composer from './composer';
 import * as npm from './npm';
 import * as nuget from './nuget';
 import * as dep from './dep';
 
+const _debug = debug('dependencies');
+
+const dependencyManagers = {
+  npm,
+  composer,
+  nuget,
+  dep,
+};
+
+const languageToFileType = {
+  JavaScript: 'npm',
+  TypeScript: 'npm',
+  PHP: 'composer',
+  'C#': 'nuget',
+  'Go': 'dep',
+};
+
+// Assumes dependencyObject is {core: ['dependency', ...], ...}
+// or {core: {dependency: ...}, ...}
+// Returns [{type: 'npm', name: 'dependency', core: 1}, ...]
+function transformToStats (manager, ...dependencyObjects) {
+  const index = {};
+  dependencyObjects.forEach( dependencies => {
+    Object.entries(dependencies).forEach(([ dependencyType, dependencyNames ]) => {
+      if (dependencyNames) {
+        if (!Array.isArray(dependencyNames)) { // Transform {dependency: ...} into ['dependency', ...]
```

Comment:
It would be a bit simpler (2 lines of code), but then it complicates the code for `composer` and `npm`. My idea was to minimize the work for package manager implementers.

You still want me to accept only `[name...]` and not `{name...}`?

Reply:
Yes, I prefer. It's not about the 2 lines of codes, it's more about the clearer contract when using the function.

Reply:
No problem, done!